diff --git a/app/controllers/trees_controller.rb b/app/controllers/trees_controller.rb
index a887e93..61da602 100644
--- a/app/controllers/trees_controller.rb
+++ b/app/controllers/trees_controller.rb
@@ -94,7 +94,7 @@ class TreesController < ApplicationController
   
   protected
     def set_xsendfile_headers(real_path, user_path, content_type = "application/x-gzip")
-      response.headers["X-Sendfile"] = File.join(GitoriousConfig["archive_cache_dir"], real_path)
+      response.headers["X-Accel-Redirect"] = GitoriousConfig["nginx_sendfile_dir"] + real_path
       response.headers["Content-Type"] = content_type
       user_path = user_path.gsub("/", "_").gsub('"', '\"')
       response.headers["Content-Disposition"] = "Content-Disposition: attachment; filename=\"#{user_path}\""
